source/Classes/1.DscResources/07.nxService.ps1

```diff
@@ -35,18 +35,25 @@ class nxService
     {
         $nxService = Get-nxService -Name $this.Name
         $currentState = [nxService]::new()
+        if ([string]::IsNullOrEmpty($nxService.Name))
+        {
+            if ($null -ne $this.Enabled -and -not $this.Enabled -and $null -ne $this.State -and $this.State -eq [nxServiceState]::Stopped)
+            {
+                return $currentState
+            }
+
+            $currentState.Reasons = [Reason]@{
+                Code = '{0}:{0}:NotRunning' -f 'nxService'
```

Comment:
This reason should say that the service does not exist. Not just that it isn't running (makes it sound like the service is there and installed, but in a Stopped state)

Reply:
The command that we are using treats both cases the same.

1. When the resource isn't running (Enabled=False and State=Stopped)
2. When the service doesn't exist and is not installed.

In both cases, "systemctl list-units --type=service --no-legend --all --no-pager \<SERVICENAME\>" will return empty output (at least on Ubuntu 20). I updated the text to "Service ''{0}'' is not running **or does not exist**." Let me know what you think.